oauthenticator/cilogon.py

```diff
@@ -273,7 +273,7 @@ def user_info_to_username(self, user_info):
             user_info_keys = sorted(user_info.keys())
             if len(claimlist) < 2:
                 self.log.error(
-                    f"Username claim {user_info_keys} not found in response: {self.username_claim}"
+                    f"Username claim {claimlist} not found in response: {user_info_keys}"
```

Comment:
I'm not confident about this code, but it seems off that `username claim` (singular) will be followed by a list of some kind.

Should it be this?

```suggestion
                    f"Username claim {self.username_claim} not found in response: {user_info_keys}"
```

But besides that, it seems very odd that the check is `if len(claimlist) < 2`. What makes that relate to if a username claim was found in the response or not?

Reply:
The username claim can either be `self.username_claim` + `self.additional_claims` (this used to exist before self.allowed_idps was introduced)
https://github.com/jupyterhub/oauthenticator/blob/462275b05d84b79966978a3c2ae1f9c2d48acd08/oauthenticator/cilogon.py#L253-L256

Or whatever the specific idp claim in `self.allowed_idps` if that's specified
https://github.com/jupyterhub/oauthenticator/blob/462275b05d84b79966978a3c2ae1f9c2d48acd08/oauthenticator/cilogon.py#L261-L265

So, claimlist can either be a list with one claim or with multiple claims to check if one fails.

I agree the `if len(claimlist) < 2` check is pretty unhappy and confusing, and the different error messages don't make a lot fo sense. I will add a commit to turn them into one.